templates/skeleton/.npmrc

```diff
@@ -1,2 +1,7 @@
 @shopify:registry=https://registry.npmjs.com
 progress=false
+
+# Ensure Vite can optimize these deps in PNPM
+public-hoist-pattern[]=cookie
+public-hoist-pattern[]=set-cookie-parser
+public-hoist-pattern[]=content-security-policy-builder
```

Comment:
Reading about [`public-hoist-pattern`](https://pnpm.io/npmrc#public-hoist-pattern) and I'm curious to know the error you encountered to know that these needed to get hoisted

Reply:
It's this issue: https://github.com/Shopify/hydrogen/issues/1834#issuecomment-2049601503

Vite can't find `cookie` to optimize it even if we tell it to do it. The reason is that if you check `node_modules/cookie`, it's not there in PNPM. It is in `node_modules/.pnpm/node_modules/cookie` instead, but Vite doesn't know that. 

This change forces pnpm to hoist `cookie` to `node_modules/cookie` so that Vite can find it.